### PR TITLE
Adds Flavourtext

### PR DESCRIPTION
**Pull request by VictorVow**
## About The Pull Request
Adds flavourtext, a way of setting a description for your character when they're examined (note it is hidden if your identity is hidden too). 

## Why It's Good For The Game
A solid roleplay feature for describing your character better than a 32x32 sprite can show. 

## Changelog
:cl:
add: Flavortext can now be set to be shown to anyone who examines you. 
/:cl:

Ports:
https://github.com/WaspStation/WaspStation-1.0/pull/173 (ported originally from Citadel)

### CODE REVIEW

File: code/datums/dna.dm [hunk 1]
```diff
@@ -320,6 +321,7 @@
 //Do not use force_transfer_mutations for stuff like cloners without some precautions, otherwise some conditional mutations could break (timers, drill hat etc)
 	if(newfeatures)
 		dna.features = newfeatures
+		flavor_text = dna.features["flavor_text"] //Update the flavor_text to use new dna text
```
Comment (zeskorion) [inline code/datums/dna.dm:324]: OH GOD NO WHY OH FUCK JESUS HOLY SHIT WHY THE FUCK IS FLAVOR TEXT A DNA FEATURE THIS IS CURSED AS FUCK

**Comment by Cenrus** [comment]
Interesting but can you make it a config option? Downstreams or the Golden station might want to opt out

**Comment by francinum** [comment]
image

**Comment by TeomanTheGreat** [comment]
We are slowly becoming Paradise Station 😔

**Comment by ike709** [comment]
...why would you do this?

**Comment by Naevii** [comment]
MRP needs something like this.

File: code/datums/dna.dm (hunk 1 above)
Comment (VictorVow) [reply code/datums/dna.dm:324]: I don't know, does it matter?

**Comment by VictorVow** [comment]
> can you make it a config option?

I can look into it, but it being an optional feature I'm not sure if there's much need to. Golden players will either leave it blank or just use this text area for memes, so it's not like it'll bother them.

File: code/datums/dna.dm (hunk 1 above)
Comment (zeskorion) [reply code/datums/dna.dm:324]: Because holy shit that's just... why

**Comment by ghost** [comment]
NO

**Comment by TotallyNotChief** [comment]
no

**Comment by r1ks-iwnl** [comment]
Maybe only add it for Sage, i kinda like it but citadel bad

**Comment by St0rmC4st3r** [comment]
I agree, make it a config option and enable only for sage, or you will traumatize poor golden players.

File: code/datums/dna.dm (hunk 1 above)
Comment (St0rmC4st3r) [reply code/datums/dna.dm:324]: I think the main reason is changelings

**Comment by ghost** [comment]
please enable flavor text for golden so we can lynch anyone who tries to RP with it

File: code/datums/dna.dm (hunk 1 above)
Comment (VictorVow) [reply code/datums/dna.dm:324]: Ah, yeah that makes sense

**Comment by Bestparty** [comment]
mfw bee becomes citadel......

**Comment by qwertyquerty** [comment]
Will put to vote after it's a config option

**Review by qwertyquerty — CHANGES_REQUESTED** [review]
Must be a config

**Comment by mehmetipekpatates** [comment]
تبا من فضلك

**Comment by VictorVow** [comment]
Will open again once I've done this

**Comment by ghost** [comment]
> Will open again once I've done this
any progress?
@Victor239

**Pull request closed without merging**